metadata-service/services/src/main/java/com/linkedin/metadata/service/DataProductService.java

```diff
@@ -58,11 +61,26 @@ public DataProductService(@Nonnull EntityClient entityClient, @Nonnull GraphClie
    * @return the urn of the newly created DataProduct
    */
   public Urn createDataProduct(
-      @Nullable String name, @Nullable String description, @Nonnull Authentication authentication) {
+      @Nullable String id,
+      @Nullable String name,
+      @Nullable String description,
+      @Nonnull Authentication authentication) {
 
     // 1. Generate a unique id for the new DataProduct.
     final DataProductKey key = new DataProductKey();
-    key.setId(UUID.randomUUID().toString());
+    if (id != null && !id.isBlank()) {
+      key.setId(id);
+    } else {
+      key.setId(UUID.randomUUID().toString());
+    }
+    try {
+      if (_entityClient.exists(
+          EntityKeyUtils.convertEntityKeyToUrn(key, DATA_PRODUCT_ENTITY_NAME), authentication)) {
+        throw new IllegalArgumentException("This Data product already exists!");
+      }
+    } catch (RemoteInvocationException e) {
+      throw new RuntimeException("Unable to check for existence of Data Product!");
```

Comment:
let's include the underlying context in this exception:

throw new RuntimeException("Unable to check for existence of Data Product!", e)